### ISSUE TITLE
scope of quickfix/location lists search on CursorMoved too wide

### ISSUE THREAD

**Issue by ptrcarta** [ervandew/eclim#462]
When installed through the arch package, eclim sets an autocmd on CursorMoved to search for error at current line, even outside of an eclipse project.
This conflicts with any other use of the quickfix or location buffers, for example vimgrep. As soon as one of these lists become big, the navigation in any file becomes terribly jerky.

as a temporary workaround I set the autocmd to only match "*.java" files, as I use eclim only on java projects, but this seems less than ideal.

to recap:
Problem: autocmd makes navigation of any file, when qf list is big, slow and irresponsive
How to reproduce: navigate to a repository with a few files in it. open vim. `:vimgrep /.*/j` . try to navigate one of the files

**Comment by ptrcarta** [comment]
https://github.com/ervandew/eclim/blob/master/org.eclim.core/vim/eclim/plugin/eclim.vim#L412

this is where the autocmd is defined. I don't know what possible solution would be best. Maybe defining the autocmds in every ftplugin at the cost of repetition, or maybe having this autocmd fire only on files defined in the path of open eclipse projects
